iOS/DuckDuckGo/DuckPlayer/DuckPlayerNavigationHandler.swift

```diff
@@ -765,25 +819,68 @@ extension DuckPlayerNavigationHandler: DuckPlayerNavigationHandling {
             return .notHandled(.duplicateNavigation)
         }
         
+        // Ensure all media playback is allowed by default
+        self.toggleMediaPlayback(webView, pause: false)
+
         // Check if DuckPlayer feature is enabled
         guard isDuckPlayerFeatureEnabled else {
             return .notHandled(.featureOff)
         }
         
-        guard let url = webView.url, let (videoID, _) = url.youtubeVideoParams else {
+        guard let url = webView.url, let (videoID, _) = url.youtubeVideoParams else {            
+            duckPlayer.dismissPill()
             return .notHandled(.invalidURL)
         }
         
-        guard url.isYoutubeWatch else {
+        guard url.isYoutubeWatch else {            
+            duckPlayer.dismissPill()
             return .notHandled(.isNotYoutubeWatch)
         }
-        
+
+        // Rest of the function remains the same
         guard videoID != lastWatchInYoutubeVideo else {
             lastURLChangeHandling = Date()
             return .handled(.newVideo)
         }
         
         let parameters = getDuckPlayerParameters(url: url)
+            
+        // Present Bottom Sheet (Native entry point) with delay
+        if duckPlayer.settings.mode == .alwaysAsk && duckPlayer.settings.nativeUI {
+            
```

Comment:
This routine is concerning to me a bit.
the `toggleMediaPlayback` will run ~20x on a spam of 1 second. If for some reason it's time consuming it could be a performance issue. I tested on profiler and it looks OK, but another reason to double check the JS code


Probably a nit but
`func handleURLChange(webView: WKWebView) -> DuckPlayerNavigationHandlerURLChangeResult`

Given the amount of parallel tasks, when this function returns it doesn't necessarily means it's done running. What do you think about making the entire function async so it's clear on the callsite? Then you could remove the Tasks

Reply:
Good call. Moved to its own async method